src/Dafda/Configuration/ConsumerOptions.cs

```diff
@@ -135,6 +135,17 @@ public void WithIncomingMessageFactory(IIncomingMessageFactory incomingMessageFa
             _builder.WithIncomingMessageFactory(incomingMessageFactory);
         }
 
+        /// <summary>
+        /// If the <see cref="IIncomingMessageFactory"/> throws an exception during message deserialization, 
+        /// Dafda will create a <see cref="TransportLevelPoisonMessage"/> 
+        /// that can be handled by the consumer instead of throwing an exception.
+        /// Note, if you wish to overwrite the default <see cref="IIncomingMessageFactory"/> you should do so before enabling poison message handling
+        /// </summary>
+        public void WithPoisonMessageHandling()
```

Comment:
So if I understand this correctly, this would also require a user to register a message handler for `transport-level-poison-message`s - would it be worth requiring a handler for this call so that it's not likely to be misconfigured?

Reply:
It would require the user to either register a message handler for transport-level-poison-message or use the "UseNoOpHandler" IUnconfiguredMessageHandlingStrategy.

Reply:
Seems to me like we could avoid misconfiguration by setting a handler here, defaulting to the no-op handler if one is not provided. Then a user ~can't~ is unlikely to create an invalid configuration :)

Reply:
This is a bit tricky, as handlers are topic specific, but poison message handling is general for the consumer